### ISSUE TITLE
callback is not called by reset(cb) and more.

### ISSUE THREAD

**Issue by yadimon** [mccormicka/Mockgoose#189]
The callbacks does not work by me.
its not called at least by reset and create
and reset cannot be called without a callback.

pseudo code:

```javascript
describe('callback', function todoDescribe() {
  const Mongoose = require('mongoose').Mongoose;
  let mockgoose = require('mockgoose');
  let mongoose = new Mongoose();
  mockgoose(mongoose);
  mongoose.connect('mongodb://localhost/mydb');

  it('should call callback by reset', (done) => {
    mockgoose.reset(done);
  });

  it('should call callback by model create', (done) => {
    const SomeSchemaFactory= mongoose.Schema;
    const SomeSchema= new SomeSchemaFactory({
      item1: String,
      item2: String,
    });
    const MyModel= mongoose.model('MyModel', SomeSchema);
    const someDoc = {
      item1: String,
      item2: String,
    };
    MyModel.create(someDoc, done);
  });
  
  it('should work with empty callback', ()=>{
    mockgoose.reset();
  });
});
```

**Comment by winfinit** [comment]
@yadimon i see that reset function without a callback was producing an error, so i've fixed that, rest of the reported issues i believe are due to your connect call. you have to wait for a connection to be established

```javascript
"use strict";


describe('callback', function todoDescribe() {
  const Mongoose = require('mongoose').Mongoose;
  let mockgoose = require('../Mockgoose');
  let mongoose = new Mongoose();
  mockgoose(mongoose);
  
  before(function(done) {
    mongoose.connect('mongodb://localhost/mydb', function() {
        done(); 
    });
  });

  it('should call callback by reset', (done) => {
    mockgoose.reset(function(err) {
       done();
    });
  });

  it('should call callback by model create', (done) => {
    const SomeSchemaFactory= mongoose.Schema;
    const SomeSchema= new SomeSchemaFactory({
      item1: String,
      item2: String,
    });
    const MyModel= mongoose.model('MyModel', SomeSchema);
    const someDoc = {
      item1: String,
      item2: String,
    };
    MyModel.create(someDoc, done);
  });

  it('should work with empty callback', ()=>{
    mockgoose.reset();
  });
});
```

**Comment by yadimon** [comment]
hm.. by me even by mongoose.connect() no callback was run  :/

tested with
"mockgoose": "5.2.4",
"mongoose": "4.0.7" and "mongoose": "4.4.3"

what could be the problem?

**Comment by winfinit** [comment]
can you please run your test with environment variable "DEBUG=*" and share your output (I would recommend running same test that you've provided in this thread

**Comment by yadimon** [comment]
ok, the problem appears if any mongod.exe instance already running. (cant use port 27017)
If i kill the process, test works. But the process is still in memory then. And i cannot runt any test  again.

last part of output:
```
Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.
    at null.<anonymous> (**\mocha\lib\runnable.js:215:19)
Tue, 16 Feb 2016 17:23:14 GMT mocha:runner run suite 
Tue, 16 Feb 2016 17:23:14 GMT mocha:runner run suite callback
Tue, 16 Feb 2016 17:23:14 GMT Mockgoose attempting to start server on port: 27017
Tue, 16 Feb 2016 17:23:14 GMT mongodb-prebuilt bin path: **\mockgoose\node_modules\mongodb-prebuilt\dist\3.2.0\bin\
Tue, 16 Feb 2016 17:23:14 GMT mongodb-prebuilt spawn **\mockgoose\node_modules\mongodb-prebuilt\dist\3.2.0\bin\mongod --storageEngine ephemeralForTest --port 27017 --dbpath **\mockgoose\.mongooseTempDB\27017
Tue, 16 Feb 2016 17:23:14 GMT mongodb 2016-02-16T18:23:14.772+0100 I CONTROL  [main] Hotfix KB2731284 or later update is not installed, will zero-out data files
Tue, 16 Feb 2016 17:23:14 GMT mongodb 2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] MongoDB starting : pid=37944 port=27017 dbpath=**\mockgoose\.mongooseTempDB\27017 64-bit host=D*
Tue, 16 Feb 2016 17:23:14 GMT mongodb 2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] targetMinOS: Windows Vista/Windows Server 2008
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] db version v3.2.0
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] git version: 45d947729a0315accb6d4f15a6b06be6d9c19fe7
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] allocator: tcmalloc
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] modules: none
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] build environment:
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten]     distarch: x86_64
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten]     target_arch: x86_64
2016-02-16T18:23:14.774+0100 I CONTROL  [initandlisten] options: { net: { port: 27017 }, storage: { dbPath: "**\mockgoose\.mongooseTempDB\27017", engine: "ephemeralForTest" } }
Tue, 16 Feb 2016 17:23:14 GMT mongodb 2016-02-16T18:23:14.775+0100 E NETWORK  [initandlisten] listen(): bind() failed errno:10048 Normalerweise darf jede Socketadresse (Protokoll, Netzwerkadresse oder Anschluss) nur jeweils einmal verwendet werden. for socket: 0.0.0.0:27017
Tue, 16 Feb 2016 17:23:14 GMT mongodb 2016-02-16T18:23:14.775+0100 E STORAGE  [initandlisten] Failed to set up sockets during startup.
2016-02-16T18:23:14.775+0100 I CONTROL  [initandlisten] dbexit:  rc: 48
Tue, 16 Feb 2016 17:23:14 GMT mongodb-prebuilt child process exited with code 48
Tue, 16 Feb 2016 17:23:24 GMT mocha:runner finished running
Tue, 16 Feb 2016 17:23:24 GMT mocha:runner end
Tue, 16 Feb 2016 17:23:24 GMT mongodb Shutting down
```

**Comment by winfinit** [comment]
@yadimon this issue is resolved in master branch, however not published to npmjs, because i am waiting for another dependency to get resolved

**Comment by winfinit** [comment]
actually, never mind, just published 5.3.0, please try it out

**Comment by yadimon** [comment]
Looks like it works!
Thank you!
